components/script/dom/html/htmldetailselement.rs

```diff
@@ -252,10 +406,49 @@ impl VirtualMethods for HTMLDetailsElement {
             .unwrap()
             .attribute_mutated(attr, mutation, can_gc);
 
-        if attr.local_name() == &local_name!("open") {
+        // Step 1. If namespace is not null, then return.
+        if *attr.namespace() != ns!() {
+            return;
+        }
+
+        // Step 2. If localName is name, then ensure details exclusivity by closing the given element if needed
+        // given element.
+        if attr.local_name() == &local_name!("name") {
+            let old_name: DOMString = match mutation {
+                AttributeMutation::Set(old) => old
+                    .map(|value| value.to_string().into())
+                    .unwrap_or_default(),
+                AttributeMutation::Removed => attr.value().to_string().into(),
+            };
+
+            if let Some(shadow_root) = self.containing_shadow_root() {
+                shadow_root
+                    .details_name_groups()
+                    .unregister_details_element(old_name, self);
+                if matches!(mutation, AttributeMutation::Set(_)) {
+                    shadow_root
+                        .details_name_groups()
+                        .register_details_element(self);
+                }
+            } else if self.upcast::<Node>().is_in_a_document_tree() {
+                let document = self.owner_document();
+                document
+                    .details_name_groups()
+                    .unregister_details_element(old_name, self);
+                if matches!(mutation, AttributeMutation::Set(_)) {
+                    document
+                        .details_name_groups()
+                        .register_details_element(self);
+                }
```

Comment:
Can we reduce duplication here by retrieving the relevant `details_name_groups` from the rootnode and then perform the steps? Given that both return `RefMut`, I would think that it should be possible to do that, is it not?

Reply:
I don't think that will work because something has to be keeping the `RefMut` alive and that's either a shadow root or a document. I don't see a way to generalize here.